src/libraries/System.Private.CoreLib/src/System/Text/ASCIIUtility.cs

```diff
@@ -351,20 +460,46 @@ private static unsafe nuint GetIndexOfFirstNonAsciiByte_Sse2(byte* pBuffer, nuin
             // instead be the second mask. If so, skip the entire first mask and drain ASCII bytes
             // from the second mask.
 
-            if (currentMask == 0)
+            if (Sse2.IsSupported)
             {
-                pBuffer += SizeOfVector128;
-                currentMask = secondMask;
+                if (!ContainsNonAsciiByte_Sse2(currentSseMaskOrAdvSimdIndex))
+                {
+                    pBuffer += SizeOfVector128;
+                    currentSseMaskOrAdvSimdIndex = secondSseMaskOrAdvSimdIndex;
+                }
+            }
+            else if (AdvSimd.IsSupported)
+            {
+                if (!ContainsNonAsciiByte_AdvSimd(currentSseMaskOrAdvSimdIndex))
+                {
+                    pBuffer += SizeOfVector128;
+                    currentSseMaskOrAdvSimdIndex = secondSseMaskOrAdvSimdIndex;
+                }
+            }
+            else
+            {
+                throw new PlatformNotSupportedException();
             }
-
         FoundNonAsciiDataInCurrentMask:
 
```

Comment:
Same for this label, it seems like duplicate computations are being made.

Reply:
Hmmm, not really. Both the SSE and AdvSimd paths can come here, so we need to special case this label for both paths